### DISCUSSION TITLE
Is there a way to Programmatically pull current note's front matter or title?

### DISCUSSION THREAD

**Discussion by rossoj85** [javalent/obsidian-leaflet#282] [Ideas]
I am trying to incorporate leaflet into a daily journal template to track where I am writing each day.  Each entry's file name will be the date.  

Instead of manually writing the file's name into the coordinates and markerFile, I want to be able to insert a variable equal to the files name to automatically set the map.  I tried with templater, but the leaflet block seems to be taking the template variables as string literals. 

The following code block might give a better idea of what I am trying to do : 

`
---
location: [40.26828968323852,-72.6091354494659]

---

<%+ tp.file.title %>

```leaflet
id: leaflet-map
coordinates: [[Location test]]
minZoom: 1
maxZoom: 18
width: 100%
defaultZoom: 15
unit: meters
scale: 1
markerFile:[<%+ tp.file.title %>]
darkMode: true
```

`

**Comment by valentine195** [comment]
So you’re going to have a map with one marker on it (the current note) for each day?

Let me think… maybe the easiest way would be supplying an empty markerFile would look in the current note.

**Comment by rossoj85** [comment]
Thanks!  I didn't realize that empty markerFile and coordinates would refer to the front matter. That worked perfectly.

**Reply by valentine195** [reply]
Oh it worked? Lol. I was talking about *making* it work that way, but I guess I was two steps ahead of myself 😂

**Reply by rossoj85** [reply]
Yea, it works great! I put the geolocation in the front matter and wrote the leaflet block like this. Maybe just add the method to the docs! 

```leaflet
id: leaflet-map
coordinates: []
minZoom: 1
maxZoom: 18
width: 100%
defaultZoom: 15
unit: meters
scale: 1
markerFile:[]
darkMode: true
```

![image](https://user-images.githubusercontent.com/25422645/153775034-11a228f8-2f42-4396-9deb-c12911f69fd1.png)
![image](https://user-images.githubusercontent.com/25422645/153775041-12f57323-230a-4edd-aed2-66e1178252bf.png)

**Comment by maradotwebp** [comment]
The method mentioned doesn't work anymore in recent versions - Setting `coordinates: []` & `markerFile: []` neither centers the map on the coordinates of the current file's frontmatter, nor does it create a marker there.

Is there another possible workaround for this?

**Reply by rossoj85** [reply]
This is true. I have notice this as well. If there is a workaround, I haven't found one.

**Comment by thiagocoutinhor** [comment]
After much tinkering I achieved this using [meta-bind](https://github.com/mProjectsCode/obsidian-meta-bind-plugin) plugin (that requires [js-engine](https://github.com/mProjectsCode/obsidian-js-engine-plugin) to achieve this, so install both):

<pre>
```meta-bind-js-view
{location} as location
---
const x = context.bound.location[0]
const y = context.bound.location[1]
const map = `\`\`\`leaflet
id: map-id
image: [[image.jpg]]
bounds:
  - [0, 0]
  - [8012, 8012]
coordinates: [${x}, ${y}]
width: 250px
height: 250px
\`\`\``
return engine.markdown.create(map)
```
</pre>

Since you will have everything installed, you can use the meta-bind-embed feature to create a template map and embed it in your notes as I did. [See the documentation](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/metabindembed/#syntax).
